src/Microsoft.Windows.CsWin32/Generator.cs

```diff
@@ -3158,17 +3161,34 @@ private ParameterSyntax CreateParameter(MethodSignature<TypeSyntax> methodSignat
                     elementType = IntPtrTypeSyntax;
                 }
 
-                // private struct __TheStruct_Count
+                var lengthLiteralSyntax = LiteralExpression(SyntaxKind.NumericLiteralExpression, Literal(length));
+
+                // internal struct __TheStruct_Count
                 // {
-                //     private TheStruct _0, _1, _2, _3, _4, _5, _6, _7, _8;
+                //     internal TheStruct _0, _1, _2, _3, _4, _5, _6, _7, _8;
+                //     /// <summary>Always <c>8</c>.</summary>
+                //     internal int Length => 8;
                 // ...
                 IdentifierNameSyntax fixedLengthStructName = IdentifierName($"__{fieldName}_{length}");
                 var fixedLengthStruct = StructDeclaration(fixedLengthStructName.Identifier)
                     .AddModifiers(Token(this.Visibility))
                     .AddMembers(
                         FieldDeclaration(VariableDeclaration(elementType)
                             .AddVariables(Enumerable.Range(0, length).Select(n => VariableDeclarator($"_{n}")).ToArray()))
-                            .AddModifiers(Token(this.Visibility)));
+                            .AddModifiers(Token(this.Visibility)),
+                        PropertyDeclaration(PredefinedType(Token(SyntaxKind.IntKeyword)), "Length")
+                            .AddModifiers(Token(this.Visibility))
+                            .WithExpressionBody(ArrowExpressionClause(lengthLiteralSyntax))
+                            .WithSemicolonToken(Token(SyntaxKind.SemicolonToken))
+                            .WithLeadingTrivia(Trivia(DocumentationCommentTrivia(SyntaxKind.SingleLineDocumentationCommentTrivia).AddContent(
+                                DocCommentStart,
+                                XmlElement("summary", List(new XmlNodeSyntax[]
+                                {
+                                    XmlText("Always "),
+                                    XmlElement("c", List(new XmlNodeSyntax[] { XmlText(lengthLiteralSyntax.Token.ValueText) })),
+                                    XmlText("."),
+                                })),
+                                DocCommentEnd))));
```

Comment:
You might not need the doc comment once we make it a constant, since I think Intellisense will show the constant's value.
Also, the precedent for doc comments is to write it out mostly as a string literal and ask the OM to parse it, since as we see here, this is super-complicated code for a comment.